src/cycax/cycad/engines/cycax_part_freecad.py

```diff
@@ -224,43 +225,73 @@ def hole(
-        """This method will be used for creating a png of 2 specific views of the object.
+
+    def render_to_png(self, view: str|None = None):
+        """Used to create a png of the desired side.
+        
         Args:
-            target_path: The Path the png needs to be saved under.
+            view: The side of the object the png will be produced from.
+            
         """
         active_doc = FreeCADGui.activeDocument()
-        target_image_file = f"{target_path}-perspectiveAll.png"
-        FreeCADGui.SendMsgToActiveView("ViewFit")
-        active_doc.activeView().viewAxometric()
-        active_doc.activeView().fitAll()
-        active_doc.activeView().saveImage(str(target_image_file), 2000, 1800, "White")
-        target_image_file = f"{target_path}-perspectiveTop.png"
+        view = view.upper()
+
+        if view is None:
+            view = "ALL"
+
+        self.change_view(active_doc=active_doc, side=view)
         FreeCADGui.SendMsgToActiveView("ViewFit")
-        active_doc.activeView().viewTop()
+
+        target_image_file = f"{self.filepath}-perspective{view}.png"
         active_doc.activeView().fitAll()
         active_doc.activeView().saveImage(str(target_image_file), 2000, 1800, "White")
 
-    def render_to_dxf(self, active_doc: App.Document, target_path: Path):
-        """This method will be used for creating a pdxf of the object currently in view.
+    def change_view(self, active_doc, side: str):
+        """This will change the gui view to show the specified side."""
+        
+        match side:
+            case "TOP":
+                active_doc.activeView().viewTop()
+            case "BACK":
+                active_doc.activeView().viewRear()
+            case "REAR":
+                active_doc.activeView().viewRear()
+            case "BOTTOM":
+                active_doc.activeView().viewBottom()
+            case "FRONT":
+                active_doc.activeView().viewFront()
+            case "LEFT":
+                active_doc.activeView().viewLeft()
+            case "RIGHT":
+                active_doc.activeView().viewRight()
+            case "ALL":
+                active_doc.activeView().viewAxometric()
+
+
+    def render_to_dxf(self, view: str|None = None):
+        """This method will be used for creating a dxf of the object currently in view.
         Args:
             active_doc: The active FreeCAD Gui
-            target_path: The Path the png needs to be saved under.
         """
+        view = view.upper()
+        active_doc = FreeCADGui.activeDocument()
+        if view is None:
+            view = "TOP"
+        self.change_view(active_doc=active_doc, side=view)
+        FreeCADGui.SendMsgToActiveView("ViewFit")
         __objs__ = []
         __objs__.append(active_doc.getObject("Shape"))
 
-        importDXF.export(__objs__, str(f"{target_path}-perspective.dxf"))
+        importDXF.export(__objs__, str(f"{self.filepath}-perspective{view}.dxf"))
```

Comment:
DXF is definitely not a perspective view. In all cases I thing the word perspective should be removed. I missed that in other PR.